lib/book/book_page/book_page_view.dart

```diff
@@ -5,66 +5,243 @@ import 'package:book/book/bloc/book_page_event.dart';
+                  context
+                      .read<BookPageBloc>()
+                      .add(AddBookPageEvent(page: result));
+                }
+              },
+              child: Icon(Icons.add),
+            ),
+            body: pageBody(context),
+            bottomSheet: pages.isNotEmpty
+                ? SafeArea(
+                    child: Container(
+                      height: kToolbarHeight,
+                      child: Row(
+                        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                        children: [
+                          IconButton(
+                            padding: const EdgeInsets.only(left: 16),
+                            onPressed: currentPageIndex > 0
+                                ? () {
+                                    if (currentPageIndex > 0) {
+                                      context.read<BookPageBloc>().add(
+                                          PreviousPageEvent(
+                                              pageIndex: currentPageIndex));
+                                    }
+                                  }
+                                : null,
+                            icon: Icon(Icons.arrow_back),
+                          ),
+                          if (pages.isNotEmpty)
+                            Container(
+                              alignment: Alignment.center,
+                              height: 30,
+                              width: 30,
+                              decoration: BoxDecoration(
+                                color: AppColors.primaryColor,
+                                shape: BoxShape.circle,
+                              ),
+                              child: Text(
+                                pages[currentPageIndex].pageNumber.toString(),
+                                textAlign: TextAlign.center,
+                                style: TextStyle(fontSize: 16),
+                              ),
+                            ),
+                          IconButton(
+                            padding: const EdgeInsets.only(right: 16),
+                            onPressed: currentPageIndex < pages.length - 1
+                                ? () {
+                                    if (currentPageIndex < pages.length - 1) {
+                                      context.read<BookPageBloc>().add(
+                                          NextPageEvent(
+                                              pageIndex: currentPageIndex));
+                                    }
+                                  }
+                                : null,
+                            icon: Icon(Icons.arrow_forward),
+                          ),
+                        ],
+                      ),
+                    ),
+                  )
+                : null,
+          );
+        } else {
+          return Center(
+            child: CircularProgressIndicator(),
```

Comment:
I don't think you need Loading and Loaded State if you you build ProgressIndicator when the state is other than DisplayBookPage State.